graylog2-server/src/main/java/org/graylog2/streams/StreamDTO.java

```diff
@@ -194,4 +224,127 @@ public StreamDTO build() {
             return autoBuild();
         }
     }
+
+    @Override
+    @JsonIgnore
+    public String getId() {
+        return id();
+    }
+
+    @Override
+    @JsonIgnore
+    public String getTitle() {
+        return title();
+    }
+
+    @Override
+    @JsonIgnore
+    public String getDescription() {
+        return description();
+    }
+
+    @Override
+    @JsonIgnore
+    public Boolean getDisabled() {
+        return disabled();
+    }
+
+    @Override
+    @JsonIgnore
+    public String getContentPack() {
+        return contentPack();
+    }
+
+    @Override
+    @JsonIgnore
+    public List<String> getCategories() {
+        return categories();
+    }
+
+    @Override
+    @JsonIgnore
+    public String getCreatorUserId() {
+        return creatorUserId();
+    }
+
+    @Override
+    @JsonIgnore
+    public DateTime getCreatedAt() {
+        return createdAt();
+    }
+
+    @Override
+    @JsonIgnore
+    public Set<ObjectId> getOutputIds() {
+        final Set<ObjectId> outputs = new HashSet<>();
+        if (outputIds() != null) {
+            outputs.addAll(outputIds());
+        }
+        return outputs;
+    }
+
+    @Override
+    @JsonIgnore
+    public Set<Output> getOutputs() {
+        final Set<Output> outputs = new HashSet<>();
+        if (outputObjects() != null) {
+            outputs.addAll(outputObjects());
+        }
+        return outputs;
+    }
+
+    @Override
+    @JsonIgnore
+    public IndexSet getIndexSet() {
+        return indexSet();
+    }
+
+    @Override
+    @JsonIgnore
+    public Boolean isPaused() {
+        return disabled();
+    }
+
+    @Override
+    @JsonIgnore
+    public List<StreamRule> getStreamRules() {
+        final List<StreamRule> rules = new ArrayList<>();
+        if (rules() != null) {
+            rules.addAll(rules());
+        }
+        return rules;
+    }
+
+    @Override
+    @JsonIgnore
+    public MatchingType getMatchingType() {
+        if (matchingType() == null) {
+            return MatchingType.AND;
+        } else {
+            return matchingType();
+        }
+    }
+
+    @Override
+    @JsonIgnore
+    public boolean isDefaultStream() {
+        return Boolean.TRUE.equals(isDefault());
+    }
+
+    @Override
+    @JsonIgnore
+    public boolean getRemoveMatchesFromDefaultStream() {
+        return Boolean.TRUE.equals(removeMatchesFromDefaultStream());
+    }
+
+    @Override
+    @JsonIgnore
+    public String getIndexSetId() {
+        return indexSetId();
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id(), title(), creatorUserId());
+    }
```

Comment:
Do we really need this? Auto-value takes care of `hashCode` and `equals` implementations.

Reply:
@bernd https://github.com/Graylog2/graylog2-server/blob/master/graylog2-server/src/test/java/org/graylog2/streams/StreamListFingerprintTest.java#L100 so this test checks for streams fingerprinting to be consistent. Previously, `StreamImpl` used `PersistedImpl#hashCode` https://github.com/Graylog2/graylog2-server/blob/master/graylog2-server/src/main/java/org/graylog2/database/PersistedImpl.java#L108C1-L110C6 to be consistent. When relying on AutoValue, the fingerprint changes every time the test is run. Only place I see the fingerprint used is in `StreamRouter` https://github.com/Graylog2/graylog2-server/blob/master/graylog2-server/src/main/java/org/graylog2/streams/StreamRouter.java#L69C1-L71C6

Reply:
The fingerprint is supposed to change only when stream routing-related fields change, so the stream router can decide when to reload the engine. The new `StreamImpl#hashCode` implementation looks good! But I think we should introduce a `StreamImpl#toStreamListFingerprint()` method (incl. Javadoc) and not override the auto-value generated `#hashCode` method. That makes the use case explicit. We should do the same when we refactor the stream rule and output objects to use mongojack.

What do you think?

Reply:
Makes a lot of sense to me!